### PR TITLE
Add stain normalization with Reinhard's method

### PR DESCRIPTION
**Pull request by alessiamarcolini**
## Description

Add stain normalization with Reinhard's method

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Fixes: #322

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/histolab/histolab/blob/master/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: histolab/stain_normalizer.py [hunk 1]
```diff
@@ -318,3 +318,104 @@ def _ordered_stack(stain_matrix: np.ndarray, order: List[int]) -> np.ndarray:
...
+            original = img_lab[:, :, i].copy()
+            new = norm_lab[:, :, i].copy()
+            original[np.not_equal(~mask[:, :, 0], True)] = 0
+            new[~mask[:, :, 0]] = 0
+            norm_lab[:, :, i] = new + original
+
+        norm_rgb = LabToRgb()(norm_lab)
+        return norm_rgb
+
+    def mean_std(self, img_rgb: PIL.Image.Image) -> Tuple[np.ndarray, np.ndarray]:
```
Comment (ernestoarbitrio) [inline histolab/stain_normalizer.py:382]: this should by private and I'd change the name to `_summary_measures`

**Review by ernestoarbitrio — CHANGES_REQUESTED** [review]
just 1 comment

File: histolab/stain_normalizer.py (hunk 1 above)
Comment (alessiamarcolini) [reply histolab/stain_normalizer.py:382]: I'd call it `_summary_statistics`

**Review by ernestoarbitrio — APPROVED** [review]

File: histolab/stain_normalizer.py [hunk 2]
```diff
@@ -318,3 +318,106 @@ def _ordered_stack(stain_matrix: np.ndarray, order: List[int]) -> np.ndarray:
...
+        ----------
+        target_rgb : PIL.Image.Image
+            Target image for stain normalization. Can be either RGB or RGBA.
+        """
+        means, stds = self._summary_statistics(target_rgb)
+        self.target_means = means
+        self.target_stds = stds
+
+    def transform(self, img_rgb: PIL.Image.Image) -> PIL.Image.Image:
+        """Normalize staining of ``img_rgb``.
```
Comment (nicolebussola) [inline histolab/stain_normalizer.py:350]: Normalize image staining

File: histolab/stain_normalizer.py [hunk 3]
```diff
@@ -318,3 +318,106 @@ def _ordered_stack(stain_matrix: np.ndarray, order: List[int]) -> np.ndarray:
...
+        mask = self._tissue_mask(img_rgb)
+
+        img_lab = RgbToLab()(img_rgb)
+        mean_per_channel = img_lab.mean(axis=(0, 1), where=mask)
+        std_per_channel = img_lab.std(axis=(0, 1), where=mask)
+        return mean_per_channel, std_per_channel
+
+    @staticmethod
+    def _tissue_mask(img_rgb: PIL.Image.Image) -> np.ndarray:
+        """Return a binary mask of the tissue in ``img_rgb``.
```
Comment (nicolebussola) [inline histolab/stain_normalizer.py:408]: , after masking tissue

File: histolab/stain_normalizer.py [hunk 4]
```diff
@@ -318,3 +318,106 @@ def _ordered_stack(stain_matrix: np.ndarray, order: List[int]) -> np.ndarray:
...
+            new[~mask[:, :, 0]] = 0
+            norm_lab[:, :, i] = new + original
+
+        norm_rgb = LabToRgb()(norm_lab)
+        return norm_rgb
+
+    def _summary_statistics(
+        self, img_rgb: PIL.Image.Image
+    ) -> Tuple[np.ndarray, np.ndarray]:
+        """Return mean and standard deviation of each channel in LAB color space.
```
Comment (nicolebussola) [inline histolab/stain_normalizer.py:385]: Statistics are computed on the masked tissue

File: histolab/stain_normalizer.py [hunk 5]
```diff
@@ -318,3 +318,106 @@ def _ordered_stack(stain_matrix: np.ndarray, order: List[int]) -> np.ndarray:
...
+
+        Parameters
+        ----------
+        img_rgb : PIL.Image.Image
+            Input image. Can be either RGB or RGBA.
+
+        Returns
+        -------
+        np.ndarray
+            Binary tissue mask,
```
Comment (nicolebussola) [inline histolab/stain_normalizer.py:418]: .

**Pull request merged**